src/main/java/com/conveyal/r5/analyst/TravelTimeSurfaceComputer.java

```diff
@@ -0,0 +1,179 @@
+                }
+            }
+        } else {
+            // we always walk to egress from transit, but we may have a different access mode.
+            // if the access mode is also walk, the pointset's linkage cache will return two references to the same
+            // linkage below
+            // TODO use directMode? Is that a resource limiting issue?
+            // also gridcomputer uses accessMode to avoid running two street searches
+            LinkedPointSet linkedDestinationsAccess = destinations.link(network.streetLayer, accessMode);
+            LinkedPointSet linkedDestinationsEgress = destinations.link(network.streetLayer, StreetMode.WALK);
+
+            if (!request.request.directModes.equals(request.request.accessModes)) {
+                LOG.warn("Disparate direct modes and access modes are not supported in analysis mode.");
+            }
+
+            // Perform street search to find transit stops and non-transit times.
+            StreetRouter sr = new StreetRouter(network.streetLayer);
+            sr.streetMode = accessMode;
+            sr.profileRequest = request.request;
+            sr.distanceLimitMeters = 2000; // TODO hardwired same as gridcomputer
+            sr.setOrigin(request.request.fromLat, request.request.fromLon);
+            sr.dominanceVariable = StreetRouter.State.RoutingVariable.DISTANCE_MILLIMETERS;
+            sr.route();
+
+            // Get the travel times to all stops reached in the initial on-street search. Convert distances to speeds.
+            int offstreetTravelSpeedMillimetersPerSecond = (int) (request.request.getSpeed(accessMode) * 1000);
+
+            TIntIntMap accessTimes = sr.getReachedStops();
+            for (TIntIntIterator it = accessTimes.iterator(); it.hasNext(); ) {
+                it.advance();
+                // TODO how to handle kiss and ride/park and ride? Clearly this is not right.
+                it.setValue(it.value() / offstreetTravelSpeedMillimetersPerSecond);
+            }
+
+            // Create a new Raptor Worker.
+            FastRaptorWorker worker = new FastRaptorWorker(network.transitLayer, request.request, accessTimes);
+
+            // Run the main RAPTOR algorithm to find paths and travel times to all stops in the network.
+            int[][] transitTravelTimesToStops = worker.route();
+
+            // find non-transit times
+            int[] nonTransitTravelTimesToDestinations = linkedDestinationsAccess
+                    .eval(sr::getTravelTimeToVertex, offstreetTravelSpeedMillimetersPerSecond)
+                    .travelTimes;
+
+            PerTargetPropagater perTargetPropagater = new PerTargetPropagater(transitTravelTimesToStops, nonTransitTravelTimesToDestinations, linkedDestinationsEgress, request.request, 120 * 60);
+            perTargetPropagater.propagateTimes((target, times) -> {
+                Arrays.sort(times);
+                int[] results = new int[nSamples];
```

Comment:
Again, I think the variable name should be changed and there should be some comment here explaining we're reading off percentiles and not reporting the entire list of travel times.